src/pipeline_manager.py

```diff
@@ -87,15 +88,35 @@ def train(pipeline_name, dev_mode, logger, params, seed):
     meta_train = meta[meta['is_train'] == 1]
     meta_valid = meta[meta['is_valid'] == 1]
 
+    train_mode = True
+
     meta_valid = meta_valid.sample(int(params.evaluation_data_sample), random_state=seed)
 
     if dev_mode:
         meta_train = meta_train.sample(20, random_state=seed)
         meta_valid = meta_valid.sample(10, random_state=seed)
 
+    if 'lgbm' in pipeline_name:
+        annotation_file_path = os.path.join(params.data_dir, 'train', "annotation.json")
+        coco = COCO(annotation_file_path)
+        meta_train = meta_train.sample(params.lgbm__num_training_examples, random_state=seed)
+        train_mode = False
+        annotations = []
+        for image_id in meta_train['ImageId'].values:
```

Comment:
@apyskir this logic should be somewhere else and only high level function should be called in the manager. So I would refactor to have something like load_lgbm_data that produces meta_train_lgbm, annotations_lgbm .

Reply:
@jakubczakon OK, you're right